packagesFromOverlaysBuilderConstructor.nix

```diff
@@ -1,92 +1,81 @@
+          ```
+          overlays = {
+          "unstable/firefox" = prev: final: {
+          firefox = prev.override { privacySupport = true; };
+          }; 
+          }
+
+          self.packages = {
+          firefox = *firefox derivation with privacySupport*;
+          }
+          ```
 
           **/
         let
 
           inherit (flake-utils-plus.lib) flattenTree filterPackages;
-          inherit (builtins) attrNames mapAttrs listToAttrs attrValues concatStringSep concatMap any;
+          inherit (builtins) foldl' attrNames mapAttrs listToAttrs
+            attrValues concatStringSep concatMap any head;
           nameValuePair = name: value: { inherit name value; };
-          filterAttrs = pred: set:
-            listToAttrs (concatMap (name: let v = set.${name}; in if pred name v then [ (nameValuePair name v) ] else [ ]) (attrNames set));
-          hasPrefix =
-            pref:
-            str: builtins.substring 0 (builtins.stringLength pref) str == pref;
 
-
-          # first, flatten and filter on valid packages (by nix flake check criterion)
           flattenedPackages =
-            let
-              f = system: tree: (filterPackages system (flattenTree tree));
-            in
-            mapAttrs f pkgs;
+            # merge all channels into one package set
+            foldl' (a: b: a // b) { } (attrValues channels);
 
-          # second, flatten all overlays' packages' keys into a single list
-          flattendOverlaysNames =
+          # flatten all overlays' packages' keys into a single list
+          flattenedOverlaysNames =
             let
-              allOverlays = concatMap (c: c.overlays) (attrValues channels);
+              allOverlays = attrValues overlays;
 
               overlayNamesList = overlay:
                 attrNames (overlay null null);
             in
-            concatMap (o: overlayNamesList o) allOverlays;
+            concatMap overlayNamesList allOverlays;
+
+          # create list of single-attribute sets that contain each package
+          exportPackagesList = map
+            (name:
+              { ${name} = flattenedPackages.${name}; }
+            )
+            flattenedOverlaysNames;
+
+          # fold list into one attribute set
+          exportPackages = foldl' (a: b: a // b) { } exportPackagesList;
+
+          system = (head (attrValues channels)).system;
 
         in
-        # finally, only retain those packages defined by overlays
-          # pkgs'.<system>.<prefix/flattend/tree/attributes>
-          # overlays' = [ "prefix" ... ];
-        mapAttrs
-          (_: pkgs:
-            filterAttrs
-              (pkgName:
-                any (overlayName: hasPrefix pkgName overlayName) flattendOverlaysNames
-              )
-              pkgs
-          )
-          flattenedPackages;
+        # flatten any nested sets with with "/" delimiter
+          # filter out packages that don't meet the flake check conditions
```

Comment:
Why doesn't nixpkgs-fmt pick up these comments alignment :/

Reply:
actually `nixpkgs-fmt` forces that formatting, I tried to put them both on the same indent, but fmt kept staggering them. I condensed it to just one comment.